service/src/main/java/uk/gov/hmcts/reform/fpl/service/NoticeOfProceedingsService.java

```diff
@@ -1,29 +1,145 @@
+                    .build()
+            ));
+        }
+
+        return listAndLabel;
+    }
+
+    public NoticeOfProceedings setNoticeOfProceedingJudge(CaseData caseData) {
+        NoticeOfProceedings noticeOfProceedings = caseData.getNoticeOfProceedings();
+
+        JudgeAndLegalAdvisor judgeAndLegalAdvisor = getSelectedJudge(
+            noticeOfProceedings.getJudgeAndLegalAdvisor(),
+            caseData.getAllocatedJudge()
+        );
+
+        removeAllocatedJudgeProperties(judgeAndLegalAdvisor);
+        noticeOfProceedings.setJudgeAndLegalAdvisor(judgeAndLegalAdvisor);
+
+        return noticeOfProceedings;
+    }
+
+    public List<Element<DocumentBundle>> uploadAndPrepareNoticeOfProceedingBundle(
+        CaseData caseData, List<DocmosisTemplates> docmosisTemplatesTypes) {
+        DocmosisNoticeOfProceeding templateData = noticeOfProceedingsTemplateDataGenerationService
+            .getTemplateData(caseData);
+
+        List<DocmosisDocument> noticeOfProceedingDocuments =
+            buildNoticeOfProceedingDocuments(templateData, docmosisTemplatesTypes);
+
+        List<Document> documentReferences = uploadDocuments(noticeOfProceedingDocuments);
+
+        return createNoticeOfProceedingsCaseData(documentReferences);
+    }
+
+    public List<Element<DocumentBundle>> prepareNoticeOfProceedingBundle(
+        List<Element<DocumentBundle>> updatedNoticeOfProceedings,
+        List<Element<DocumentBundle>> noticeOfProceedingBundleBefore,
+        List<DocmosisTemplates> templateTypes) {
 
-    public List<Element<DocumentBundle>> getRemovedDocumentBundles(CaseData caseData,
+        if (isNotEmpty(noticeOfProceedingBundleBefore)) {
+            updatedNoticeOfProceedings.addAll(getRemovedDocumentBundles(noticeOfProceedingBundleBefore, templateTypes));
+        }
+
+        return updatedNoticeOfProceedings;
+    }
+
+    public List<Element<DocumentBundle>> getPreviousNoticeOfProceedings(CaseData caseDataBefore) {
+        if (caseDataBefore == null || caseDataBefore.getNoticeOfProceedingsBundle() == null) {
+            return List.of();
+        }
+
+        return caseDataBefore.getNoticeOfProceedingsBundle();
+    }
+
+    private List<DocmosisDocument> buildNoticeOfProceedingDocuments(DocmosisNoticeOfProceeding templateData,
                                                                    List<DocmosisTemplates> templateTypes) {
+        return templateTypes.stream()
+            .map(template -> docmosisDocumentGeneratorService.generateDocmosisDocument(templateData, template))
+            .collect(Collectors.toList());
+    }
+
+    private List<Document> uploadDocuments(List<DocmosisDocument> documents) {
+        return documents.stream()
+            .map(document -> uploadDocumentService.uploadPDF(document.getBytes(), document.getDocumentTitle()))
+            .collect(Collectors.toList());
+    }
+
+    private List<Element<DocumentBundle>> createNoticeOfProceedingsCaseData(List<Document> uploadedDocuments) {
```

Comment:
maybe not the best name, method takes in list of documents and returns list of element document bundle, theres no mention of case data or notice of proceedings, maybe buildDocumentBundleList()

Reply:
looks like DocumentBundle is only used for NoP 🤔